app/models/query/results/group_by.rb

```diff
@@ -89,11 +89,50 @@ def transform_custom_field_keys(groups)
 
     if custom_field.list?
       transform_list_custom_field_keys(custom_field, groups)
+    elsif custom_field.field_format_hierarchy?
+      transform_hierarchy_custom_field_keys(custom_field, groups)
     else
       transform_single_custom_field_keys(custom_field, groups)
     end
   end
 
+  def transform_hierarchy_custom_field_keys(custom_field, groups)
+    items = hierarchy_items_for_keys(custom_field, groups)
+
+    groups.transform_keys do |key|
+      if custom_field.multi_value?
+        (key ? key.split(".") : []).map do |subkey|
```

Comment:
`String#split` always returns an array. We are doing a `truthy` check if `falsey` we are going for an empty array.

We could probably do something on the lines of

```ruby
Array(key&.split(".")).map
```